examples/defaulter-gen/generators/defaulter.go

```diff
@@ -458,20 +469,24 @@ func buildCallTreeForType(t *types.Type, root bool, existingDefaulters, newDefau
 					name = field.Type.Name.Name
 				}
 			}
-			if child := buildCallTreeForType(field.Type, false, existingDefaulters, newDefaulters); child != nil {
+			if child := buildCallTreeForType(field.Type, false, existingDefaulters, newDefaulters, alreadyVisitedTypes); child != nil {
 				child.field = name
 				parent.children = append(parent.children, *child)
 			}
 		}
 	case types.Alias:
-		if child := buildCallTreeForType(t.Underlying, false, existingDefaulters, newDefaulters); child != nil {
+		if child := buildCallTreeForType(t.Underlying, false, existingDefaulters, newDefaulters, alreadyVisitedTypes); child != nil {
 			parent.children = append(parent.children, *child)
 		}
 	}
 	if len(parent.children) == 0 && len(parent.call) == 0 {
 		//glog.V(6).Infof("decided type %s needs no generation", t.Name)
 		return nil
 	}
+
+	// The type now acts as a parent, not a nested recursive type.
+	// We can now build the tree for it safely.
+	alreadyVisitedTypes[t] = false
```

Comment:
The name is wrong if you set it to false after you are done. Should be "currentlyBuildingTypes" or "typesInProgress" or something.